### PR TITLE
Added finer gesture controls and parameters

### PR DESCRIPTION
**Pull request by InstrinsicAutomations**
Added an update timer on a 16ms interval. Added 3 parameters for finer gesture controls: 
* `swipeCoef` - "power". Any value < 1.0 reduces the delta value calculated from the swipe.
* `dampCoef` - "friction". Clamps between 0.001 (hard-stop) and 0.999 (float).
* `maxSpeed` - After the 2 parameters above are calculated, the rotation speed will be limited to this max rotation speed.

This PR also fixes a parsing problem I encountered.

### CODE REVIEW

File: lib/object_3d.dart
Comment (arnovanliere) [inline lib/object_3d.dart:19]: You should probably assert this is between 0 and 1 as well.

File: lib/object_3d.dart
Comment (arnovanliere) [inline lib/object_3d.dart:49]: This Timer has to be cancelled in the `dispose` method

File: lib/object_3d.dart
Comment (arnovanliere) [inline lib/object_3d.dart]: Al these types are not necessary. These are inferred when using `final adx = _deltaX.abs();` for example

File: lib/object_3d.dart
Comment (arnovanliere) [inline lib/object_3d.dart]: These lines can also be combined immediately to make it a bit shorter

```suggestion
        _deltaX = math.min(maxSpeed, adx) * sx * dampCoef;
        _deltaY = math.min(maxSpeed, ady) * sy * dampCoef;
```

File: lib/object_3d.dart
Comment (arnovanliere) [inline lib/object_3d.dart]: This type is not necessarye

File: lib/object_3d.dart
Comment (arnovanliere) [inline lib/object_3d.dart]: This one neither. `final iV = vertices[face[0] - 1];`  suffices

**Review by arnovanliere — CHANGES_REQUESTED** [review]
When running the example (which is present in this repo), i noticed that left-to-right and right-to-left swipes work exactly opposite of what they should. Should be fixed as well before merge

File: lib/object_3d.dart
Comment (TheMaverickProgrammer) [reply lib/object_3d.dart:19]: The code clamps before use.

Comment (InstrinsicAutomations) [reply lib/object_3d.dart:19]: This value is clamped before use between `0.001` and `0.999`

File: lib/object_3d.dart
Comment (InstrinsicAutomations) [reply lib/object_3d.dart:49]: noted thanks

File: lib/object_3d.dart
Comment (InstrinsicAutomations) [reply lib/object_3d.dart]: I'm not a fan of omitting types. They cause more issues than the short-term convenience addresses. If you want them gone I can do that for you.

File: lib/object_3d.dart
Comment (InstrinsicAutomations) [reply lib/object_3d.dart]: see: https://github.com/arnovanliere/object_3d/pull/2#discussion_r1523723082

I can omit this for you.

File: lib/object_3d.dart
Comment (InstrinsicAutomations) [reply lib/object_3d.dart]: https://github.com/arnovanliere/object_3d/pull/2#discussion_r1523723082

File: lib/object_3d.dart
Comment (arnovanliere) [reply lib/object_3d.dart:19]: Yes I saw, but the user doesn't get feedback if an invalid value is provided which will then be clamped between 0.001 and 0.999 and might wonder why their code doesn't work. If you assert the value range and provide an error message, the user of this library will not be able to provide an invalid value.

File: lib/object_3d.dart
Comment (arnovanliere) [reply lib/object_3d.dart]: It would be more consistent with the already existing code of this library, but whatever. I'm not going to block this merge because of this.

File: lib/object_3d.dart
Comment (InstrinsicAutomations) [reply lib/object_3d.dart:19]: In that same vein, do you want me to remove the exception catch block? This way all errors are thrown and the user can handle them. Also a controller would be nice to have here for these sorts of things. e.g. `objectController.hasError()` or something.

File: lib/object_3d.dart
Comment (InstrinsicAutomations) [reply lib/object_3d.dart:49]: done

File: lib/object_3d.dart
Comment (InstrinsicAutomations) [reply lib/object_3d.dart]: done.

File: lib/object_3d.dart
Comment (InstrinsicAutomations) [reply lib/object_3d.dart]: done.

File: lib/object_3d.dart
Comment (InstrinsicAutomations) [reply lib/object_3d.dart]: done.

File: lib/object_3d.dart
Comment (arnovanliere) [reply lib/object_3d.dart:19]: Yeah a Widget controller is a common thing in Flutter, so if you wanted to implement this, go ahead. If not, we can also merge this PR with just the assertion. Up to you

File: lib/object_3d.dart
Comment (arnovanliere) [inline lib/object_3d.dart]: Although this repo does not have a linter set up, this is something the linter would complain about. Local variables do not have to be made private be prefixing them with `_` since they are already local.

File: lib/object_3d.dart
Comment (InstrinsicAutomations) [reply lib/object_3d.dart:19]: I want to but let's get this PR in first so we have incremental changes.

File: lib/object_3d.dart
Comment (InstrinsicAutomations) [reply lib/object_3d.dart]: noted

Comment (InstrinsicAutomations) [reply lib/object_3d.dart]: done

**Comment by InstrinsicAutomations** [comment]
Replaced value clamping with requested asserts. Requesting re-review.

**Review by arnovanliere — APPROVED** [review]

**Pull request merged**
